applications/StructuralMechanicsApplication/custom_constitutive/generic_small_strain_isotropic_plasticity.cpp

```diff
@@ -86,13 +86,12 @@ void GenericSmallStrainIsotropicPlasticity<TConstLawIntegratorType>::CalculateMa
             BaseType::CalculateCauchyGreenStrain( rValues, r_strain_vector);
         }
 
-        if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) ) {
-            BaseType::CalculateElasticMatrix( r_constitutive_matrix, rValues);
-        }
-
-        if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_STRESS ) ) {
+        if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_STRESS ) ||
+            r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR )) {
             Vector& r_stress_vector = rValues.GetStressVector();
             if( r_constitutive_law_options.Is( ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR ) ) {
+                Matrix& r_constitutive_matrix = rValues.GetConstitutiveMatrix();

```

Comment:
this is actually wrong. you should NEVER use the tangent in computing the stress since the tangent is NOT secant (definitely not for the complete stress increment)

stress should be always computed by the exact function

BaseType::CalculatePK2Stress

and the tangent computed separetely

Reply:
OK, in my defense was already implemented that way

Reply:
Look at the master

https://github.com/KratosMultiphysics/Kratos/blob/4ee7f236fcd45b50dd34e3e62f29ab82f926baac/applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp#L80

Reply:
i am not telling you it is your fault... but this implementation is wrong, as it relies on r_constitutive_matrix to be the secant matrix, which is not the expected behaviour (it should be ideally the tangent)